src/Symfony/Component/Console/Helper/QuestionHelper.php

```diff
@@ -242,9 +243,10 @@ private function autocomplete(OutputInterface $output, Question $question, $inpu
             } elseif ("\177" === $c) { // Backspace Character
                 if (0 === $numMatches && 0 !== $i) {
                     --$i;
-                    $fullChoice = self::substr($fullChoice, 0, $i);
                     // Move cursor backwards
-                    $output->write("\033[1D");
+                    $output->write(sprintf("\033[%dD", (new UnicodeString(self::substr($fullChoice, -1)))->width(false)));
```

Comment:
self::substr() could be replaced by a method on the string object
but there is something to fix before: what if $fullChoice is not UTF-8?
We might need a new named constructor in the String component, that'd return ByteString|UnicodeString depending on the UTF8-compliancy of the input.

Reply:
I thought we would replace the content of those helper methods by calls to the String component since they are use multiples times, and since they are a public API. That's something to do in the next "iterations" to me. I mean it's unrelated to the bug.

Reply:
This breaks the commands on terminals that don't use UTF-8...

Reply:
> `throw new InvalidArgumentException('Invalid UTF-8 string.');`

Ok I see what you mean, sorry 😅